File: euler10.hs
{-Summation of primes
Problem 10

The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million.-}



ldp :: Integer -> Integer
ldp n = ldpf primes1 n


ldpf :: [Integer] -> Integer -> Integer
ldpf (p:ps) n | rem n p ==0 = p
              | p^2 > n     = n
              | otherwise   = ldpf ps n

primes1 :: [Integer]
primes1 = 2 : filter prime [3..1999999]

prime :: Integer -> Bool
prime n | n < 1     = error "not a positive integer"
        | n == 1    = False
        | otherwise = ldp n == n

result = sum primes1

main = do
  print result

File: cppeuler/euler10.cpp
// euler10.cpp
// Summation of primes
// Problem 10
// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

// Find the sum of all the primes below two million.

#include <iostream>
#include <vector>
#include "./tools.h"

#define sievelimit 2000000

int main() {
  long long sum = 0;
  std::vector<int> ps = primeSieve(sievelimit);
  for (int i = 0; i<sievelimit; i++) {
    sum += ps[i];
  }
  std::cout << sum << std::endl;
}
